sway-ir/src/parser.rs

```diff
@@ -59,16 +60,27 @@ mod ir_builder {
 
             rule contract() -> IrAstModule
                 = "contract" _ "{" _
-                  configs:init_config()* fn_decls:fn_decl()* "}" _
+                  configs:init_config()* _ global_vars:global_var()* _ fn_decls:fn_decl()* "}" _
                   metadata:metadata_decls() {
                     IrAstModule {
                         kind: crate::module::Kind::Contract,
                         configs,
+                        global_vars,
                         fn_decls,
                         metadata
                     }
                 }
 
+            rule global_var() -> IrAstGlobalVar
+                = m:("mut" _)? _ "global" _ name:path() _ ":" _ ty:ast_ty() _ "=" _ init:(op_const())? {
```

Comment:
Even if we don't have an initializer, the rule still makes the `=` being mandatory which is odd. I am here also for having it the same as with local variables. A single optional init rule that contains also the `=`.